flexmeasures/api/dev/sensors.py

```diff
@@ -186,6 +186,70 @@ def get(self, id: int, asset: GenericAsset):
         attributes = ["name", "timezone", "timerange_of_sensors_to_show"]
         return {attr: getattr(asset, attr) for attr in attributes}
 
+    @route("/<id>/chart_annotations", strict_slashes=False)
+    @use_kwargs(
+        {"asset": AssetIdField(data_key="id")},
+        location="path",
+    )
+    @use_kwargs(
+        {
+            "event_starts_after": AwareDateTimeField(format="iso", required=False),
+            "event_ends_before": AwareDateTimeField(format="iso", required=False),
+            "beliefs_after": AwareDateTimeField(format="iso", required=False),
+            "beliefs_before": AwareDateTimeField(format="iso", required=False),
+        },
+        location="query",
+    )
+    @permission_required_for_context("read", ctx_arg_name="asset")
+    def get_chart_annotations(self, id: int, asset: GenericAsset, **kwargs):
+        """GET from /asset/<id>/chart_annotations
+
+        .. :quickref: Chart; Download annotations for use in charts
+
+        Annotations for use in charts (in case you have the chart specs already).
+        """
+        df = get_annotations_data(
+            sensor_id=None,
+            asset_or_sensor=asset,
+            relationship_module=GenericAssetAnnotationRelationship,
+            kwargs=kwargs,
+        )
+
+        return df.to_json(orient="records")
+
+
+def get_annotations_data(
+    sensor_id: int | None,
+    asset_or_sensor: GenericAsset | Sensor,
+    relationship_module: Any,
+    kwargs,
+):
+    """
+    This function fetches a sensor or an asset annotations data
+    """
+    event_starts_after = kwargs.get("event_starts_after", None)
+    event_ends_before = kwargs.get("event_ends_before", None)
+    if asset_or_sensor is GenericAsset:
+        asset_or_sensor_class = asset_or_sensor.generic_asset
+    else:
+        asset_or_sensor_class = asset_or_sensor
+
+    df = asset_or_sensor_class.search_annotations(
```

Comment:
Why do we need the class, actually? Isn't `search_annotations` a method we can call on the instance rather than the class? It's function signature does suggest so, because its first parameter is `self` rather than `cls`.
```suggestion
    df = asset_or_sensor.search_annotations(
```